test/e2e_node/shortened_grace_period_test.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"github.com/onsi/ginkgo/v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	admissionapi "k8s.io/pod-security-admission/api"
+	"time"
+)
+
+var _ = SIGDescribe("Shortened Grace Period", func() {
+	f := framework.NewDefaultFramework("shortened-grace-period")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
+	ginkgo.Context("When repeatedly deleting pods", func() {
+		var podClient *e2epod.PodClient
+		ginkgo.BeforeEach(func() {
+			podClient = e2epod.NewPodClient(f)
+		})
+		ginkgo.It("should be deleted immediately", func() {
+			const (
+				gracePeriod      = 100
+				gracePeriodShort = 10
+			)
+			podName := "test"
+			podClient.CreateSync(getGracePeriodTestPod(podName, gracePeriod))
+			err := podClient.Delete(context.TODO(), podName, *metav1.NewDeleteOptions(gracePeriod))
+			framework.ExpectNoError(err)
+			start := time.Now()
+			podClient.DeleteSync(podName, *metav1.NewDeleteOptions(gracePeriodShort), gracePeriod*time.Second)
```

Comment:
Yes, you are right.  I'll answer your two questions and answers accordingly 
1.  The first question, the second command will override the grace period of the first command.  That's what this PR does.  And it does achieve this purpose. 
2.  Second question, do you think another verification should be done to test when the grace period is 0?  You are also right that this is indeed a current bug in kubelet, but I have found a way to fix it. 
Here are my considerations.  My original PR was intended to fix two problems.  1: shorten the grace period .2: force deletion (grace period of 0). But for some reason I put the second one on hold first.  Because the code and logic to modify these two issues are not the same.  Your comment that the API SERVER will delete the pod when the grace period is 0, but the kubelet will not, is also a bug in the code and I have found this problem.  And I was able to fix it.  But I need the current PR to finish before I can launch a new PR to solve the grace period of 0.  I don't have permission to initiate two PRs at the same time. 
Also, thank you for your concern about this PR. I will initiate a new PR to resolve the forced deletion (grace period of 0) as soon as this issue is resolved 
 
 
 
 
Translated with www.DeepL.com/Translator (free version)